roles/confluent.kafka/tasks/main.yml

```diff
@@ -1,139 +1,139 @@
-    owner: "{{kafka.broker.user}}"
-    group: "{{kafka.broker.group}}"
+    owner: "{{kafka.user}}"
+    group: "{{kafka.group}}"
   when: security_mode == "kerberos"
   notify:
     - restart kafka
 - name: broker ssl kerberos config
   template:
     src: server_kerberos_ssl.properties.j2
-    dest: "{{kafka.broker.config_file}}"
+    dest: "{{kafka.config_file}}"
     mode: 0640
-    owner: "{{kafka.broker.user}}"
-    group: "{{kafka.broker.group}}"
+    owner: "{{kafka.user}}"
+    group: "{{kafka.group}}"
   when: security_mode == "kerberos_ssl"
   notify:
     - restart kafka
 - name: broker ssl kerberos config
   template:
     src: server_kerberos_ssl.self_signed.properties.j2
-    dest: "{{kafka.broker.config_file}}"
+    dest: "{{kafka.config_file}}"
     mode: 0640
-    owner: "{{kafka.broker.user}}"
-    group: "{{kafka.broker.group}}"
+    owner: "{{kafka.user}}"
+    group: "{{kafka.group}}"
   when: security_mode == "kerberos_ssl_customcerts"
   notify:
     - restart kafka
 - name: broker Jaas config
   template:
     src: kafka_server_jaas.conf.j2
-    dest: "{{kafka.broker.jaas_file}}"
+    dest: "{{kafka.jaas_file}}"
     mode: 0640
-    owner: "{{kafka.broker.user}}"
-    group: "{{kafka.broker.group}}"
+    owner: "{{kafka.user}}"
+    group: "{{kafka.group}}"
   when: security_mode == "kerberos" or security_mode == "kerberos_ssl" or security_mode == "kerberos_ssl_customcerts"
   notify:
     - restart kafka
 - name: create systemd override file
   file:
-    path: "{{kafka.broker.systemd_override}}"
-    owner: "{{kafka.broker.user}}"
-    group: "{{kafka.broker.group}}"
+    path: "{{kafka.systemd_override}}"
+    owner: "{{kafka.user}}"
+    group: "{{kafka.group}}"
     state: directory
     mode: 0640
 - name: write environment variables
   template:
     src: environment.j2
-    dest: "{{kafka.broker.systemd_override}}/override.conf"
+    dest: "{{kafka.systemd_override}}/override.conf"
     mode: 0640
-    owner: "{{kafka.broker.user}}"
-    group: "{{kafka.broker.group}}"
+    owner: "{{kafka.user}}"
+    group: "{{kafka.group}}"
   when: security_mode == "plaintext" or security_mode == "sasl_ssl" or security_mode == "ssl_customcerts"
   notify:
     - reload systemd
     - restart kafka
 - name: write environment variables
   template:
     src: kerberos_environment.j2
-    dest: "{{kafka.broker.systemd_override}}/override.conf"
+    dest: "{{kafka.systemd_override}}/override.conf"
     mode: 0640
-    owner: "{{kafka.broker.user}}"
-    group: "{{kafka.broker.group}}"
+    owner: "{{kafka.user}}"
+    group: "{{kafka.group}}"
   when: security_mode == "kerberos" or security_mode == "kerberos_ssl" or security_mode == "kerberos_ssl_customcerts"
   notify:
     - reload systemd
     - restart kafka
 - name: broker configure service
   systemd:
-    name: "{{kafka.broker.service_name}}"
-    enabled: "{{kafka.broker.systemd.enabled}}"
-    state: "{{kafka.broker.systemd.state}}"
+    name: "{{kafka.service_name}}"
```

Comment:
Recommend leaving in 'broker' as kafka is not a component but the platform name.